src/Mvc/Mvc.Core/test/ApplicationModels/ControllerActionDescriptorProviderTests.cs

```diff
@@ -304,7 +304,7 @@ public void GetDescriptors_ActionWithHttpMethods_AddedToEndpointMetadata()
         }
 
         [Fact]
-        public void GetDescriptors_ActionWithMultipleHttpMethods_SingleHttpMethodMetadata()
+        public void GetDescriptors_ActionWithMultipleHttpMethods_LastHttpMethodMetadata()
```

Comment:
REVIEW: There's logic in [DefaultApplicationModelProvider.CreateSelectorModel()](https://github.com/dotnet/aspnetcore/blob/2f66d585c80d3273b4b5504185952bb291ee5ea0/src/Mvc/Mvc.Core/src/ApplicationModels/DefaultApplicationModelProvider.cs#L660-L671) that adds `IHttpMethodMetadata` to the endpoint given any attributes that implement `IActionHttpMethodProvider`. Given multiple `IActionHttpMethodProvider` attributes, it merges it into a single `IHttpMethodMetadata`.

This test needed to be uploaded because now the `HttpMethodAttribute` implements `IHttpMethodMetadata` and all Action attributes get added directly as metadta.

`Endpoint.Metadata.GetMetadata<IHttpMethodMetadata>()` only returns the last `IHttpMethodMetadata` and that's what gets used which happens to be the merged one. AFAICT, this is the only test that failed due to this change, but I want to make sure this doesn't have any unforeseen consequences.

@pranavk

Reply:
FYI the selectormodel concept + the way these attributes work for MVC is super complex for a few lame reasons/scenarios that don't really get used in practice.

I'd suggest trying to design the system so that these cases aren't required. The thing is that if the attributes aren't expressive enough for what you want then **worst-case** you need to write a second method that forwards to your first one.

Example of what I mean:

```C#
[Route("/foo")]
[HttpPost]
[HttpPut("/bar")]
public void MyMethod() { ... }

[HttpGet]
[HttpPost("/foo")]
[HttpPut("/bar")]
public void MyMethod() { ... }
```

What do those mean? (MVC knows).